### ISSUE TITLE
Shared memory impact on benchmarks 

### ISSUE THREAD

**Issue by jrbourbeau** [coiled/coiled-runtime#431]
In the monthly Dask community meeting today @martindurant mentioned he wanted to see what impact his shared memory PR (xref https://github.com/dask/distributed/pull/6503) has on the benchmarks here. 

@fjetter mentioned running on Coiled clusters might not have much of an impact because each worker is running on it's own VM

Opening an issue to keep the conversation going

**Comment by martindurant** [comment]
Agreed, we except to see no difference (positive or negative) where there is one process per container. However, it would be interesting to run the benchmark suite in various scenarios anyway:
- coiled
- one worker
- worker processes on the same node, one thread per worker, no shm
- worker processes on the same node, one thread per worker, with shm

where the total number of threads is a constant. I guess if this is against S3 data, the nodes should all be in the same zone for consistency. Also, it is pretty likely that the total memory use stat in the benchmarks does not take account of shm correctly.

**Comment by fjetter** [comment]
cc @crusaderky you expressed interest in this topic

**Comment by martindurant** [comment]
As a preview to my pydata talk, I did *not* find situations where applying SHM to all interprocess transfers above some size resulted in measurable benefit for example shuffly dataframe and array workloads ( https://github.com/martindurant/shm-distributed/blob/main/shm_distributed/test_runs.py#L79 ). Only scatter/broadcast gave the expected big benefit, in which case there's no need to put this complexity into the serialisation layer, it can be something done explicitly by the user at scatter time.
